qiskit/transpiler/preset_passmanagers/level0.py

```diff
@@ -94,7 +94,22 @@ def level_0_pass_manager(pass_manager_config: PassManagerConfig) -> PassManager:
     unitary_synthesis_method = pass_manager_config.unitary_synthesis_method
     unitary_synthesis_plugin_config = pass_manager_config.unitary_synthesis_plugin_config
 
-    # 1. Choose an initial layout if not set by user (default: trivial layout)
+    # 1. Decompose so only 1-qubit and 2-qubit gates remain
+    _unroll3q = [
+        # Use unitary synthesis for basis aware decomposition of UnitaryGates
+        UnitarySynthesis(
+            basis_gates,
+            approximation_degree=approximation_degree,
+            coupling_map=coupling_map,
```

Comment:
If we haven't laid out the qubits yet, what does the coupling map mean for the plugin?  We _could_ omit it, but then it starts to feel like we'll not following the calling conventions we laid out in the definition of the synthesis plugin.

Reply:
Yeah, this is a good point. If the circuit doesn't have a layout set this will be doing the wrong thing, because the synthesis pass assumes it's going to have the bits in physical order in the dag. I think we should set this to `None` if run prior to layout. `None` is still a valid value here to indicate there is no target coupling map set.

Reply:
Oh yeah, I forgot we still passed through explicit `None` on that one.  Yeah, that seems like the best solution for now.

Reply:
> If the circuit doesn't have a layout set this will be doing the wrong thing,

Any suggested test to add about this?

Reply:
It's a bit tricky to test for because it doesn't come up with the default/built-in unitary synthesis mechanism since for >=3q it uses isometry which isn't coupling map aware. For the default plugin this only comes up with <=2q unitaries. It's only for potential plugins like AQC or BQSKit where this would come up, since the synthesis pass sends bit indices to the plugin assuming their in physical order: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/passes/synthesis/unitary_synthesis.py#L272-L276 (ie post routing)

Reply:
You could set up a `unittest.mock.MagicMock` wrapper around the default synthesis plugin to catch all calls, then run the transpiler pass and assert that the first call has `min_qubits=3` and `coupling_map=None`, perhaps?  There's some examples of this ~~sort of test in `test_unitary_synthesis_plugin.py`, like here: https://github.com/Qiskit/qiskit-terra/blob/7e118e81344cb05aada5082e3cc71c1c239baaa5/test/python/transpiler/test_unitary_synthesis_plugin.py#L220-L235